### DISCUSSION TITLE
how to crop image that has a white border with bounding box

### DISCUSSION THREAD

**Discussion by srini8080** [ImageMagick/ImageMagick#6194] [Digital Image Processing]
Centos 8
Version: ImageMagick 7.1.0-35 Q16-HDRI x86_64 20062 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenMP(4.5) 
Delegates (built-in): bzlib djvu fftw fontconfig freetype heic jbig jng jp2 jpeg lzma pangocairo png tiff webp x xml zlib
Compiler: gcc (8.5)

I have a image with a black bounding box and white border. I want to only get the image without the border and the box

Thanks,

![img-000](https://user-images.githubusercontent.com/12046302/227771871-8cca7b0f-335f-4a97-bc2a-97627fff389f.png)
![img](https://user-images.githubusercontent.com/12046302/227771881-f1dc9e2b-76e8-4745-b130-5adf1d3c55c6.png)

**Comment by fmw42** [comment]
Your white background is not clean, it has tiny black spots. So removing the background is hard.  Also the black border is not nice and rectangular aligned with the axes. That makes it harder.  

This does remove all the borders but trims a little of the image.  Basically, trim the outer white, shave the black area, trim the inner white.

`magick 227771871-8cca7b0f-335f-4a97-bc2a-97627fff389f.png -fuzz 95% -trim +repage -shave 30x30 -fuzz 70% -trim +repage result.png
`

![result](https://user-images.githubusercontent.com/13458218/227788117-adfeb716-afc3-4fff-b072-e3c1d10cd451.png)

**Reply by srini8080** [reply]
I have many pictures like this so having to play with the fuzz value.

Any intelligent solution similar to your multicorp?

**Reply by fmw42** [reply]
You could try thresholding your image to binary (plus morphology) and then use connected component processing to find the bounding box of the largest dark region. Then use that bounding box to crop your original image. (The issue is to use a large enough morphology filter so that the morphology open operation connects all the important dark parts of your image after thresholding. I note that your image has lots of white space in it, so it needs a large morphology filter).

Unix syntax:

The connected components mean-color=true, allows the output after any filtering to show the mean colors of each region (black or white in this case). The keep-top=1, means to filter so that only the top region by area plus the background color are kept. The verbose=true allows the textual data for the regions to be output. I then filter with grep to keep only the black, i.e. gray(0) region and the awk prints only the bounding box for that region.


```
bbox=`im7 magick 227771871-8cca7b0f-335f-4a97-bc2a-97627fff389f.png \
-auto-threshold otsu \
-morphology open disk:11 \
-type bilevel \
-define connected-components:mean-color=true \
-define connected-components:keep-top=1 \
-define connected-components:verbose=true \
-connected-components 8 null: |\
grep "gray(0)" | awk '{print $2}'`
echo $bbox
im7 magick 227771871-8cca7b0f-335f-4a97-bc2a-97627fff389f.png \
-crop $bbox +repage \
result.png
```

![result](https://user-images.githubusercontent.com/13458218/228325433-a0e9b480-127d-4f99-afa9-fe4aa9cec4a8.png)

**Reply by srini8080** [reply]
great!. This worked perfect. Thanks much
